### DISCUSSION TITLE
How to extend existing Controller

### DISCUSSION THREAD

**Discussion by zimbar** [wintercms/winter#901] [Q&A] [answered]
Hy,

I want to add new function to the controller  "Backend\Controllers\Users". How can i do this using a plugin.

Thanks

**Comment by LukeTowers** [comment, marked as answer]
See https://wintercms.com/docs/services/behaviors#dynamically-creating-methods. 

```php
\Backend\Controllers\Users::extend(function ($controller) {
    $controller->addDynamicMethod('myMethod', function () use ($controller) {
        // do stuff.
    });
});
```

**Comment by zimbar** [comment]
Thanks, 

I thought that 'addDynamicMethod' work only with models. I will test

**Reply by LukeTowers** [reply]
It works with any class that includes the ExtendableTrait or extends the base Extendable class
